classification_model_v2.py

```diff
@@ -0,0 +1,217 @@
+
+    def lemmatize_doc(self,doc):
+        return [t.lemma_ for t in doc if self.keep_token(t)]
+
+    def prepare_docs(self,input_list):
+
+        nlp = spacy.load('en_core_web_sm')
+
+        token_vocab=[]
+        sentences =[]
+        for text in input_list:
+            doc = nlp(text.text)
+            token_vocab.append(self.lemmatize_doc(doc))
+            for sent in doc.sents:
+                sentences.append(sent.text)
+
+        num_docs = len(token_vocab) #num_docs will be reused multiple times
+
+        """creating gensim dictionary and corpus for the nouns"""
+        dictionary = corpora.Dictionary(token_vocab)
+        # dictionary.save('/tmp/docdict.dict') #uncomment to save dict for document reanalysis
+        corpus = [dictionary.doc2bow(text) for text in token_vocab]
+        # corpora.MmCorpus.serialize('/tmp/corpus.mm',corpus) #uncomment to save corpus as Mmatrix
+        """activating tfidf analysis on corpus"""
+        tfidf = gensim.models.TfidfModel(corpus,normalize=True)
+        corpus_tfidf = tfidf[corpus]
+        self.corpus_tfidf,self.corpus,self.num_docs,self.dictionary,self.sentences = corpus_tfidf,corpus,num_docs,dictionary,sentences
+        return corpus_tfidf,corpus,num_docs,dictionary,sentences
+
+    def turn_corpus_into_useful_array(self,num_docs,corpus_to_use,dictionary):
+        """
+        with this method, we attach the original vocab word to either the tfidf score-entry or the frequency score-entry.
+        inputs:
+        For this demonstration:
+        for tfidf, var corpus_to_use = corpus_tfidf
+        for frequency, var corpus_to_use = corpus
+
+        num_docs = the number of documents in question, which for this demo has been predefined as num_docs
+
+        output:
+        np array with the requisite words and scores
+        """
+        result_with_vocab_base=[ [] for i in range(num_docs)]
+        for ind,doc in enumerate(corpus_to_use):
+            result_with_vocab_base[ind] = sorted(doc,key=lambda i: i[1], reverse=True)
+            #probably can eliminate this redundant step, but need to convert from native tuple to list for appending vocab text
+
+        result_with_vocab = [ [] for i in range(num_docs)]
+
+        for ind,doc in enumerate(result_with_vocab_base):
+            for entry in doc:
+                result_with_vocab[ind].append(list([*entry,dictionary[entry[0]]]))
+
+        result = np.array(result_with_vocab)
+
+        return result
+
+
+    def fetch_top_words(self,documentnumber, model_df,n=3):
+        """
+        this convenience function will:
+
+        output the n top-scoring words for the scoring model in question.
+
+        the function simply takes the values of the first three words for each document, so any sorted model output can be used.
+
+        """
+
+        topwords = []
+        for i in model_df[documentnumber][0:n]:
+            topwords.append(i[2])
+        return topwords
+
+    # IN DEVELOPMENT
```

Comment:
Is this still being worked on or is it ready to be merged? Can we remove the commented code?